lib/currency.rb

```diff
@@ -0,0 +1,29 @@
+module Currency
+  extend self
+  
+  def args(*currency)
+    currency = currency.flatten.compact
+    return if currency.blank?
+    case currency.length
+    when 0
+      nil
+    when 1
+      arg = currency[0]
+      return if arg.blank?
+      case arg.length
+      when 3 #e.g. 'EUR'
+        [arg] 
+      when 6 #e.g 'USDEUR' -> ['USD', 'EUR']
+        arg.scan(/.{3}/).uniq
+      else
+        raise ArgumentError.new("Wrong currency argument")
+      end
+    else
+      if currency.select{|it| it.length == 3}.length == currency.length
+        currency.uniq
+      else
+        raise ArgumentError.new("Wrong currency argument")
+      end
+    end
+  end
+end
```

Comment:
My version (it contains another possible string handling, I forgot to mention previously):

```ruby
module Currency
  # @return [Array] the list of currencies
  # @param [String+]
  def args(*currency)
    return nil if currency.length.zero? # return is OK here, as preliminary check

    # I forgot to mention, we have another weird format for currency pairs: 'EUR|USD'
    curr = if currency.length == 1 # string
              if currency.first.length.reminder(3).zero? # 'EURUSD'
                currency.first.scan /.{3}/
              elsif currency.first.length == 7           # 'EUR|USD'
                currency.split '|'
              else
                [''] # will raise an error below, DRY
              end
           else
             currency
           end
    
    # OK, now we have an array
    raise ArgumentError.new("Wrong currency argument [#{currency}]") unless curr.all? { |it| it.length == 3 }

    curr.uniq
  end
end
```